pkg/fakesocket/fake.go

```diff
@@ -19,40 +18,73 @@ var FakeAddr = &net.UDPAddr{
 	Port: 8181,
 }
 
+var ErrClosedConnection = errors.New("Connection is closed")
+var ErrAlreadyClosedConnection = errors.New("Connection is already closed")
+
 // FakePacketConn is a fake net.PacketConn providing FakeMetric when read from.
-type FakePacketConn struct{}
+type FakePacketConn struct {
+	closed chan int
```

Comment:
The convention is to use `struct {}` not empty interface - `struct {}` is a zero-sized value (as in its representation is 0 bytes of memory). Any value of interface type consumes memory of two pointers (in general; I think certain cases are/were optimized. Also I remember seeing something about removing this optimization for reasons).

It is not important in this case, but it is confusing to see interface there :)
An example where that matters is a buffered channel of `struct {}` - you can put `N` values in it but it does not consume memory to hold them, it's just a semaphore.